### PR TITLE
[#IR-309] As a dev, Added Update Password API and changed recover method to reuse it.

### PR DESCRIPTION
**Pull request by chahat-arora**

### CODE REVIEW

**Comment by himanshu-garg-165** [comment]
Please do the following:
- It looks like the OTP is not being deleted after being used, please try to reproduce it and act upon it.

File: controllers/user_auth.go [hunk 1]
```diff
@@ -54,7 +56,33 @@ func (ctrl UserAuthController) Identify(c *gin.Context) {
 
 // Recover ...
 func (ctrl UserAuthController) Recover(c *gin.Context) {
-	status, err := ctrl.AuthService.Recover(c)
+
+	var RecoveryData userSerializers.Recover
```
Comment (himanshu-garg-165) [inline controllers/user_auth.go]:
```suggestion
	var recoveryData userSerializers.Recover
```

**Review by himanshu-garg-165 — CHANGES_REQUESTED** [review]

File: controllers/user_auth.go (hunk 1 above)
Comment (chahat-arora) [reply controllers/user_auth.go]: done.

**Comment by chahat-arora** [comment]
> Please do the following:
> 
> * It looks like the OTP is not being deleted after being used, please try to reproduce it and act upon it.

fixed.

File: apps/user/services/authentication.go [hunk 2]
```diff
@@ -198,6 +193,35 @@ func (service AuthenticationService) Recover(c *gin.Context) (
...
+	tx := service.DB.Begin()
+	user := userModels.User{}
+
+	err = tx.Model(&userModels.User{}).Where("email = ?", userPasswordData.Email).Update("password", encryptedPassword).Scan(&user).Error
+	if err != nil {
+		tx.Rollback()
+		return http.StatusInternalServerError, err
+	}
+
+	err = tx.Where("user_id = ?", user.ID).Delete(&userModels.OTP{}).Error
```
Comment (himanshu-garg-165) [inline apps/user/services/authentication.go:217]: Please add proper comments here and other places.

**Review by himanshu-garg-165 — CHANGES_REQUESTED** [review]

File: apps/user/services/authentication.go (hunk 2 above)
Comment (chahat-arora) [reply apps/user/services/authentication.go:217]: done.

**Review by himanshu-garg-165 — APPROVED** [review]

**Pull request merged**